File: 9-palindrome-number/palindrome-number.c
// я здесь пробую чисто арифметическое решение, без превращения в строку
bool isPalindrome(int x) {
    if (x < 0) return false;
    long reverse_x = 0; 
    int temp_x = x;
    int i = 1;
    while (temp_x != 0) {
        reverse_x = reverse_x * 10 + temp_x % 10;
        temp_x = temp_x / 10;
    }
    return x == reverse_x;
}

File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        elif 0 <= x < 10 :
            return True
        else:
            digit = tuple(int(char) for char in str(abs(x)))
            left = 0          
            right = len(digit) - 1 
            while left < right:
                if digit[left] == digit[right]:
                    left += 1
                    right -= 1
                else:
                    return False
            return True


